crates/red_knot_python_semantic/src/types.rs

```diff
@@ -58,9 +58,7 @@ fn symbol_ty_by_id<'db>(db: &'db dyn Db, scope: ScopeId<'db>, symbol: ScopedSymb
             Some(bindings_ty(
                 db,
                 use_def.public_bindings(symbol),
-                use_def
-                    .public_may_be_unbound(symbol)
-                    .then_some(Type::Unknown),
+                use_def.public_may_be_unbound(symbol).then_some(Type::Never),
```

Comment:
below this is the case where `use_def.has_public_declarations(symbol)` is false, what would be a situation where we fall into that case?

Reply:
```python
if flag:
    x = 1
```
See my comment on #13514